### DISCUSSION TITLE
How to hide/show columns based on quick filter value?

### DISCUSSION THREAD

**Discussion by ShifanilaSenthil** [serenity-is/Serenity#6798] [Help]
I have a quick filter in the grid and have persisted grid settings set (User Preferences) then how can I hide and show the columns based on the quick filter value? I have tried the below code, but didn't help me
 if ((w as EnumEditor).value == 1){
columns = columns.filter(f => f.field != 'field1');
                    columns = columns.filter(f => f.field != 'field2');
                    columns = columns.filter(f => f.field != 'field3');
                    this.slickGrid.setColumns(columns);
}
 if ((w as EnumEditor).value == 2){
columns = columns.filter(f => f.field != 'field4');
                    columns = columns.filter(f => f.field != 'field5');
                    this.slickGrid.setColumns(columns);
}

Can somebody help me out with this?

**Comment by wezmag** [comment]
Try this in the constructor of you grid.
``` typescript
const filter = this.findQuickFilter(EnumEditor, "YourFieldName");
filter.changeSelect2(e => {
   const columns = this.allColumns;
   tryFirst(columns, x=> x.field === 'field1').visible = filter.value === 2;
   tryFirst(columns, x=> x.field === 'field2').visible = filter.value === 2;
   tryFirst(columns, x=> x.field === 'field3').visible = filter.value === 2;
   tryFirst(columns, x=> x.field === 'field4').visible = filter.value === 1;
   tryFirst(columns, x=> x.field === 'field5').visible = filter.value === 1;
   this.slickGrid.setColumns(columns);
});
```
Above code is not tested, please modify accordingly.

**Reply by ShifanilaSenthil** [reply]
Thank you for your response.

**Reply by edwardch** [reply]
@ShifanilaSenthil please consider marking this answer as "Answered" to help anyone in the community in future who might have the same or similar question.
